pkg/storage/tsdb/bucketindex/markers_bucket_client.go

```diff
@@ -128,12 +126,25 @@ func (b *globalMarkersBucket) ReaderWithExpectedErrs(fn objstore.IsOpFailureExpe
 	return b
 }
 
-func (b *globalMarkersBucket) isBlockDeletionMark(name string) (ulid.ULID, bool) {
-	if path.Base(name) != metadata.DeletionMarkFilename {
-		return ulid.ULID{}, false
+func (b *globalMarkersBucket) isMark(name string) (string, bool) {
+	marks := map[string]func(ulid.ULID) string{
+		metadata.DeletionMarkFilename:  BlockDeletionMarkFilepath,
+		metadata.NoCompactMarkFilename: NoCompactMarkFilenameMarkFilepath,
 	}
 
-	// Parse the block ID in the path. If there's not block ID, then it's not the per-block
-	// deletion mark.
-	return block.IsBlockDir(path.Dir(name))
+	for mark, globalFilePath := range marks {
+		if path.Base(name) == mark {
```

Comment:
I am a bit confused here, `mark` seems to be a function, and `path.Base()` returns a string. Is there some Go magic that gives you something meaningful comparing string to function?